Orm/Xtensive.Orm/Orm/Providers/SqlCompiler.Helpers.cs

```diff
@@ -211,6 +221,10 @@ private static bool ShouldUseQueryReference(CompilableProvider origin, SqlProvid
         .Where(i => i >= 0)
         .ToList();
       var containsCalculatedColumns = calculatedColumnIndexes.Count > 0;
+      var typeIdIsOnlyCalculatedColumn = (calculatedColumnIndexes.Count == 1)
+        ? IsTypeIdColumn(sourceSelect.Columns[calculatedColumnIndexes[0]])
+        : false;

```

Comment:
```suggestion
      var typeIdIsOnlyCalculatedColumn = (calculatedColumnIndexes.Count == 1)
        ? IsTypeIdColumn(sourceSelect.Columns[calculatedColumnIndexes[0]])
        : false;
      var typeIdIsOnlyCalculatedColulmn = calculatedColumnIndexes.Count==1 && IsTypeIdColumn(sourceSelect.Columns[calculatedColumnIndexes[0]]);
```

Reply:
Hi @alex-kulakov. Have you seen this suggestion?

Reply:
Done